src/radical/entk/appman/appmanager.py

```diff
@@ -173,6 +181,47 @@ def assign_workflow(self, workflow):
 
         self._prof.prof('assigning workflow', uid=self._uid)
         self._workflow = workflow
+       
+        pipelines = list(self._workflow)
+        self._data['pipelines'].append({'Total pipeline(s)': len(pipelines)})
+
+        for i in range(len(pipelines)):
+            pipeline = pipelines[i]
+            stages = list(pipeline.stages)
+
+            self._data['pipelines'].append({'pipeline': i, 
+                                      'pipeline name' : pipeline.name,
+                                      'pipelines uid' : pipeline.uid, 
+                                      'total stages': len(stages)})
```

Comment:
Why do you record `i`?  Also, this is a pipeline entry,  and a shorter dict would suffice:
```
{'name'  : pipeline.name,
 'uid'   : pipeline.uid,
 'stages': len(stages)
}
```

Same holds for the other dicts below.